doc/api/http.md

```diff
@@ -587,13 +587,21 @@ added: v1.6.0
 -->
 
 * `name` {string}
-* Returns: {string}
+* Returns: {string|string[]}
 
 Reads out a header on the request. Note that the name is case insensitive.
+The type of the return value depends on the arguments provided to
+[`response.setHeader()`][].
 
 Example:
 ```js
-const contentType = request.getHeader('Content-Type');
+request.setHeader('content-type', 'text/html');
+request.setHeader('Content-Length': Buffer.byteLength(body));
+request.setHeader('Set-Cookie', ['type=ninja', 'language=javascript'])
+const contentType = request.getHeader('Content-Type');      // contentType is 'text/html'
+const contentLength = request.getHeader('Content-Length');  // contentLength is of type number
```

Comment:
If this is true, should the parameter types be updated? Can we define the whole possible set there?

Reply:
Well, it's javascript so mostly everything is coerced to a string (even `undefined`). So actually `setHeader()` accepts any type and as a result `getHeader()` returns any type as it is a pass through.

Reply:
So maybe we can use `{any}` there? @Trott, what do you think?

Reply:
> So maybe we can use {any} there? @Trott, what do you think?

Works for me. Easy enough to update it to something else if we discover this confuses a lot of people. I don't think it will, though.